vars/withGoogleSA.groovy

```diff
@@ -0,0 +1,6 @@
+def call(String credentialsId, Closure body) {
+    withCredentials([file(credentialsId: credentialsId, variable: 'GC_KEY')]) {
+        sh("gcloud auth activate-service-account --key-file=${GC_KEY}")
```

Comment:
A couple of questions:
- do we need to log out later?
- Is it ok to call this function multiple times within the same pipeline?
- or in a concurrent pipeline?

Reply:
We don't need to log out, key allows you to get token for gcloud, it impersonates current user as a service account, we also have on-demand machines so I see no potential issues in this approach